pkg/dockerloginner/dockerloginner.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2017 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dockerloginner
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/nuclio/nuclio/pkg/dockerclient"
+	"github.com/nuclio/nuclio/pkg/errors"
+
+	"github.com/nuclio/nuclio-sdk"
+)
+
+// DockerLoginner initializes docker client credentials
+type DockerLoginner struct {
+	logger       nuclio.Logger
+	dockerClient dockerclient.Client
+}
+
+func NewDockerLoginner(parentLogger nuclio.Logger,
+	dockerClient dockerclient.Client) (*DockerLoginner, error) {
+
+	return &DockerLoginner{
+		logger:       parentLogger.GetChild("loginner"),
+		dockerClient: dockerClient,
+	}, nil
+}
+
+func (dl *DockerLoginner) LoginFromDir(keyDir string) error {
+	dockerKeyDirFileInfos, err := ioutil.ReadDir(keyDir)
+	if err != nil {
+		return errors.Wrap(err, "Failed to read docker key directory")
+	}
+
+	for _, dockerKeyDirFileInfo := range dockerKeyDirFileInfos {
+		if dockerKeyDirFileInfo.IsDir() {
+			continue
+		}
+
+		dockerKeyPath := path.Join(keyDir, dockerKeyDirFileInfo.Name())
+
+		dockerKeyDirFile, err := os.Open(dockerKeyPath)
```

Comment:
You don't close the file (unless I missed it:)
Also, why not use `ioutil.ReadFile` to read the file directly?

Reply:
Changed to ReadFile. D'oh